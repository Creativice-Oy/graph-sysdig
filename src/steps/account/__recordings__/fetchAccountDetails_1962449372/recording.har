{
  "log": {
    "_recordingName": "fetchAccountDetails",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "5.1.1"
    },
    "entries": [
      {
        "_id": "0d29cdc1ea9a8e8f7ab793f1e65b94a7",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "app.us4.sysdig.com"
            }
          ],
          "headersSize": 285,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://app.us4.sysdig.com/api/user/me"
        },
        "response": {
          "bodySize": 2675,
          "content": {
            "mimeType": "application/json;charset=utf-8",
            "size": 2675,
            "text": "{\"user\":{\"termsAndConditions\":true,\"timezone\":\"+00:00\",\"pictureUrl\":\"http://www.gravatar.com/avatar/ba5b87ffa4527a556b5f6da95fe36fe5\",\"customerSettings\":{\"sysdig\":{\"enabled\":false,\"enabledSSE\":false,\"buckets\":[]},\"plan\":{\"maxAgents\":15,\"onDemandAgents\":0,\"maxTeams\":-1,\"timelines\":[{\"from\":null,\"to\":null,\"sampling\":10000000},{\"from\":null,\"to\":null,\"sampling\":60000000},{\"from\":null,\"to\":null,\"sampling\":600000000},{\"from\":null,\"to\":null,\"sampling\":3600000000},{\"from\":null,\"to\":null,\"sampling\":86400000000}],\"metricsSettings\":{\"enforce\":false,\"showExperimentals\":false,\"limits\":{\"jmx\":500,\"statsd\":1000,\"appCheck\":500,\"prometheus\":8000,\"prometheusPerProcess\":500,\"connections\":80,\"progAggregationCount\":12,\"appCheckAggregationCount\":12,\"promMetricsWeight\":0.0,\"topFilesCount\":10,\"topDevicesCount\":10,\"hostServerPorts\":10,\"containerServerPorts\":5,\"limitKubernetesResources\":false,\"kubernetesPods\":10000,\"kubernetesJobs\":10000,\"containerDensity\":200,\"meerkatSuited\":true},\"enforceAgentAggregation\":true,\"enablePromCalculatedIngestion\":false,\"enablePromScrapeV2\":true,\"enableGlobalK8sAllPodContainers\":false},\"secureEnabled\":true,\"monitorEnabled\":false,\"allocatedAgentsCount\":15,\"subscriptionState\":\"trialing\",\"paymentsIntegrationId\":{\"id\":\"not present\",\"ttl\":{\"ttl\":43}},\"pricingPlan\":\"pro\",\"indirectCustomer\":false,\"trialPlanName\":\"secure-enterprise-standard-monthly\",\"partner\":\"Direct\",\"trialEndDate\":1664383443236,\"subscriptionVersion\":2,\"migratedToV2Direct\":false,\"overageAssessmentEligible\":false},\"environment\":{}},\"customer\":{\"id\":5001686,\"name\":null,\"accessKey\":\"1be2fe7d-f00f-4513-9a15-1384c35d8497\",\"externalId\":\"fbd8b3fc-781c-4e7f-afac-a2ee9309f4c9\",\"dateCreated\":1661791442836},\"oauth\":false,\"agentInstallParams\":{\"accessKey\":\"1be2fe7d-f00f-4513-9a15-1384c35d8497\",\"collectorAddress\":\"ingest.us4.sysdig.com\",\"collectorPort\":6443,\"checkCertificate\":true,\"sslEnabled\":true},\"remoteWriteParams\":{\"serverAddress\":\"app.us4.sysdig.com\"},\"properties\":{\"has_been_invited\":false},\"resetPassword\":false,\"additionalRoles\":[],\"teamRoles\":[{\"teamId\":50006626,\"teamName\":\"Secure Operations\",\"teamTheme\":\"#7BB0B2\",\"userId\":50003375,\"userName\":\"stefan.virag@contractor.jupiterone.com\",\"role\":\"ROLE_TEAM_MANAGER\",\"admin\":true},{\"teamId\":50006625,\"teamName\":\"Monitor Operations\",\"teamTheme\":\"#7BB0B2\",\"userId\":50003375,\"userName\":\"stefan.virag@contractor.jupiterone.com\",\"role\":\"ROLE_TEAM_MANAGER\",\"admin\":true}],\"lastUpdated\":1663493736587,\"dateActivated\":1661791669709,\"accessKey\":\"1be2fe7d-f00f-4513-9a15-1384c35d8497\",\"intercomUserIdHash\":\"7941192844d448aedd650df4bd26799efd5fca275a1f1eef775ee67b6cf061ad\",\"uniqueIntercomUserId\":\"50003375.fbd8b3fc-781c-4e7f-afac-a2ee9309f4c9\",\"currentTeam\":50006626,\"name\":\"stefan.virag@contractor.jupiterone.com\",\"id\":50003375,\"version\":2,\"enabled\":true,\"firstName\":\"Stefan\",\"lastName\":\"Virag\",\"systemRole\":\"ROLE_CUSTOMER\",\"username\":\"stefan.virag@contractor.jupiterone.com\",\"status\":\"confirmed\",\"products\":[\"SDC\",\"SDS\"],\"dateCreated\":1661791442936,\"lastSeenOnSecure\":1663493736587}}"
          },
          "cookies": [
            {
              "httpOnly": true,
              "name": "INGRESSCOOKIEAPI",
              "path": "/",
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "date",
              "value": "Sun, 18 Sep 2022 09:46:17 GMT"
            },
            {
              "name": "vary",
              "value": "Origin, Access-Control-Request-Method, Access-Control-Request-Headers, Accept-Encoding, User-Agent"
            },
            {
              "name": "content-type",
              "value": "application/json;charset=utf-8"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "cache-control",
              "value": "no-cache, no-store, max-age=0, must-revalidate"
            },
            {
              "name": "pragma",
              "value": "no-cache"
            },
            {
              "name": "expires",
              "value": "0"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=15768000"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 498,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-09-18T09:46:16.712Z",
        "time": 518,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 518
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
