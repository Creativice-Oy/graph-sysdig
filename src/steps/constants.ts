import {
  RelationshipClass,
  StepEntityMetadata,
  StepRelationshipMetadata,
} from '@jupiterone/integration-sdk-core';

export const Steps = {
  ACCOUNT: 'fetch-account-details',
  USERS: 'fetch-users',
  BUILD_ACCOUNT_AND_USER_RELATIONSHIP: 'build-account-and-user-relationship',
  TEAMS: 'fetch-teams',
  IMAGE_SCANS: 'fetch-image-scans',
  BUILD_ACCOUNT_AND_IMAGE_SCAN_RELATIONSHIP:
    'build-account-and-image-scan-relationship',
  BUILD_ACCOUNT_AND_TEAM_RELATIONSHIP: 'build-account-and-team-relationship',
  BUILD_TEAM_AND_USER_RELATIONSHIP: 'build-team-and-user-relationship',
};

export const Entities: Record<
  'ACCOUNT' | 'USER' | 'TEAM' | 'IMAGE_SCAN',
  StepEntityMetadata
> = {
  ACCOUNT: {
    resourceName: 'Account',
    _type: 'sysdig_account',
    _class: ['Account'],
  },
  USER: {
    resourceName: 'User',
    _type: 'sysdig_user',
    _class: ['User'],
  },
  TEAM: {
    resourceName: 'Team',
    _type: 'sysdig_team',
    _class: ['Team'],
  },
  IMAGE_SCAN: {
    resourceName: 'Image Scan',
    _type: 'sysdig_image_scan',
    _class: ['Assessment'],
  },
};

export const Relationships: Record<
  | 'ACCOUNT_HAS_USER'
  | 'ACCOUNT_HAS_TEAM'
  | 'TEAM_HAS_USER'
  | 'ACCOUNT_HAS_IMAGE_SCAN',
  StepRelationshipMetadata
> = {
  ACCOUNT_HAS_USER: {
    _type: 'sysdig_account_has_user',
    sourceType: Entities.ACCOUNT._type,
    _class: RelationshipClass.HAS,
    targetType: Entities.USER._type,
  },
  ACCOUNT_HAS_TEAM: {
    _type: 'sysdig_account_has_team',
    sourceType: Entities.ACCOUNT._type,
    _class: RelationshipClass.HAS,
    targetType: Entities.TEAM._type,
  },
  TEAM_HAS_USER: {
    _type: 'sysdig_team_has_user',
    sourceType: Entities.TEAM._type,
    _class: RelationshipClass.HAS,
    targetType: Entities.USER._type,
  },
  ACCOUNT_HAS_IMAGE_SCAN: {
    _type: 'sysdig_account_has_image_scan',
    sourceType: Entities.ACCOUNT._type,
    _class: RelationshipClass.HAS,
    targetType: Entities.IMAGE_SCAN._type,
  },
};
