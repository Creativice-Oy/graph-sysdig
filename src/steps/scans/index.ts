import {
  createDirectRelationship,
  Entity,
  IntegrationStep,
  IntegrationStepExecutionContext,
  RelationshipClass,
} from '@jupiterone/integration-sdk-core';
import { createAPIClient } from '../../client';

import { IntegrationConfig } from '../../config';
import { ACCOUNT_ENTITY_KEY } from '../account';
import { Steps, Entities, Relationships } from '../constants';
import { SCANNER_ENTITY_KEY } from '../scanner';
import { createImageScanEntity, createImageScanEntityV2 } from './converter';

export async function fetchImageScans({
  instance,
  jobState,
}: IntegrationStepExecutionContext<IntegrationConfig>) {
  const apiClient = createAPIClient(instance.config);

  await apiClient.iterateImageScans(async (scan) => {
    await jobState.addEntity(createImageScanEntity(scan));
  });

  await apiClient.iterateImageScansV2(async (scan) => {
    const imageScanDetails = await apiClient.fetchImageScansV2Details(scan.id);
    await jobState.addEntity(createImageScanEntityV2(imageScanDetails));
  });
}

export async function buildAccountAndImageScansRelationship({
  jobState,
}: IntegrationStepExecutionContext<IntegrationConfig>) {
  const accountEntity = (await jobState.getData(ACCOUNT_ENTITY_KEY)) as Entity;

  await jobState.iterateEntities(
    { _type: Entities.IMAGE_SCAN._type },
    async (imageScanEntity) => {
      if (accountEntity && imageScanEntity) {
        await jobState.addRelationship(
          createDirectRelationship({
            _class: RelationshipClass.HAS,
            from: accountEntity,
            to: imageScanEntity,
          }),
        );
      }
    },
  );
}

export async function buildScannerAndImageScansRelationship({
  jobState,
}: IntegrationStepExecutionContext<IntegrationConfig>) {
  const scannerEntity = (await jobState.getData(SCANNER_ENTITY_KEY)) as Entity;

  await jobState.iterateEntities(
    { _type: Entities.IMAGE_SCAN._type },
    async (imageScanEntity) => {
      if (scannerEntity && imageScanEntity) {
        await jobState.addRelationship(
          createDirectRelationship({
            _class: RelationshipClass.PERFORMED,
            from: scannerEntity,
            to: imageScanEntity,
          }),
        );
      }
    },
  );
}

export const scansSteps: IntegrationStep<IntegrationConfig>[] = [
  {
    id: Steps.IMAGE_SCANS,
    name: 'Fetch Image Scans',
    entities: [Entities.IMAGE_SCAN],
    relationships: [],
    dependsOn: [],
    executionHandler: fetchImageScans,
  },
  {
    id: Steps.BUILD_ACCOUNT_AND_IMAGE_SCAN_RELATIONSHIP,
    name: 'Build Account and Image Scan Relationship',
    entities: [],
    relationships: [Relationships.ACCOUNT_HAS_IMAGE_SCAN],
    dependsOn: [Steps.IMAGE_SCANS, Steps.ACCOUNT],
    executionHandler: buildAccountAndImageScansRelationship,
  },
  {
    id: Steps.BUILD_SCANNER_AND_IMAGE_SCAN_RELATIONSHIP,
    name: 'Build Scanner and Image Scan Relationship',
    entities: [],
    relationships: [Relationships.SCANNER_PERFORMED_IMAGE_SCAN],
    dependsOn: [Steps.IMAGE_SCANS, Steps.SCANNER],
    executionHandler: buildScannerAndImageScansRelationship,
  },
];
