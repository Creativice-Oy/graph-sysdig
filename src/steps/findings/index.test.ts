import {
  createMockStepExecutionContext,
  Recording,
} from '@jupiterone/integration-sdk-testing';
import { IntegrationConfig } from '../../config';
import { fetchFindings } from '.';
import { integrationConfig } from '../../../test/config';
import { setupSysdigRecording } from '../../../test/recording';
import { Relationships } from '../constants';
import { fetchImageScans } from '../scans';

describe('#fetchFindings', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupSysdigRecording({
      directory: __dirname,
      name: 'fetchFindings',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data and build relationships with image scan and cve', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchImageScans(context);
    await fetchFindings(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      collectedEntities: context.jobState.collectedEntities,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    const imageScans = context.jobState.collectedEntities.filter((e) =>
      e._type.includes('sysdig_image_scan'),
    );
    expect(imageScans.length).toBeGreaterThan(0);

    const findings = context.jobState.collectedEntities.filter((e) =>
      e._type.includes('sysdig_finding'),
    );
    expect(findings.length).toBeGreaterThan(0);
    expect(findings).toMatchGraphObjectSchema({
      _class: ['Finding'],
      schema: {
        additionalProperties: false,
        properties: {
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          _type: { const: 'sysdig_finding' },
          id: { type: 'string' },
          name: { type: 'string' },
          category: { type: 'string' },
          severity: { type: 'string' },
          numericSeverity: { type: 'number' },
          open: { type: 'boolean' },
          'vulnerability.name': { type: 'string' },
          'vulnerability.severity.reporter.name': { type: 'string' },
          'vulnerability.severity.reporter.url': { type: 'string' },
          'vulnerability.severity.value': { type: 'number' },
          'vulnerability.cvssScore.reporter.name': { type: 'string' },
          'vulnerability.cvssScore.reporter.url': { type: 'string' },
          'vulnerability.cvssScore.value.version': { type: 'string' },
          'vulnerability.cvssScore.value.score': { type: 'number' },
          'vulnerability.cvssScore.value.vector': { type: 'string' },
          'vulnerability.exploitable': { type: 'boolean' },
          'vulnerability.exploit': { type: 'string' },
          'vulnerability.description': { type: 'string' },
          'vulnerability.disclosureDate': { type: 'number' },
          'vulnerability.solutionDate': { type: 'number' },
          'package.id': { type: 'string' },
          'package.name': { type: 'string' },
          'package.version': { type: 'string' },
          'package.type': { type: 'string' },
          'package.path': { type: 'string' },
          'package.running': { type: 'string' },
          fixedInVersion: { type: 'string' },
        },
      },
    });

    expect(
      context.jobState.collectedRelationships.filter(
        (e) => e._type === Relationships.IMAGE_SCAN_IDENTIFIED_FINDING._type,
      ),
    ).toMatchDirectRelationshipSchema({
      schema: {
        properties: {
          _class: { const: 'IDENTIFIED' },
          _type: {
            const: 'sysdig_image_scan_identified_finding',
          },
        },
      },
    });

    expect(
      context.jobState.collectedRelationships.filter(
        (e) => e._type === Relationships.FINDING_IS_CVE._type,
      ),
    ).toMatchDirectRelationshipSchema({
      schema: {
        properties: {
          _class: { const: 'IS' },
          _type: {
            const: 'sysdig_image_scan_identified_finding',
          },
        },
      },
    });
  });
});
