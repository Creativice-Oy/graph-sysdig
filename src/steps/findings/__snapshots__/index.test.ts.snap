// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#fetchFindings should collect data and build relationships with image scan and cve 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "Assessment",
      ],
      "_key": "sysdig_image_scan:17144f2f45c295bc2fdcae5db51044ca",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "fixablePackages": Array [],
            "id": "17144f2f45c295bc2fdcae5db51044ca",
            "isEVEEnabled": false,
            "labels": null,
            "metadata": Object {
              "architecture": "amd64",
              "author": "",
              "baseOS": "debian 11.5",
              "createdAt": "2022-09-13T00:56:19.369186372Z",
              "digest": "debian@sha256:73060f9fdf46e142b63fddde45759f12aea093156445c32ab667338d0563f5c4",
              "imageId": "sha256:43d28810c1b4c28a1be3bac8e0e40fcc472b2bfcfcda952544ed99cb874d2b1a",
              "labels": null,
              "layersCount": 1,
              "os": "linux",
              "pullString": "debian",
              "size": 129313792,
            },
            "packageCount": 96,
            "packageTypes": Array [
              "os",
            ],
            "policyEvaluations": Array [
              Object {
                "creationTimestamp": "2022-09-02T11:58:59.03062Z",
                "evaluationResult": "failed",
                "failuresCount": Object {
                  "imageConfigCreationDate": 0,
                  "imageConfigDefaultUser": 0,
                  "imageConfigEnvVariable": 0,
                  "imageConfigInstructionIsPkgManager": 0,
                  "imageConfigInstructionNotRecommended": 0,
                  "imageConfigLabel": 0,
                  "imageConfigSensitiveInformationAndSecrets": 0,
                  "vulnDenyList": 0,
                  "vulnSeverityAndThreats": 2,
                },
                "id": "17144f2f45c295bc9785414e83306ba9",
                "identifier": "sysdig_best_practices",
                "name": "Sysdig Best Practices",
                "policyType": "alwaysApply",
                "updateTimestamp": "2022-09-02T11:58:59.03062Z",
              },
            ],
            "runningFixablePackages": null,
            "storedAt": "2022-09-13T03:53:46.858993084Z",
            "type": "dockerImage",
            "vulnsBySev": Array [
              Object {
                "fixableCount": 0,
                "totalCount": 0,
              },
              Object {
                "fixableCount": 0,
                "totalCount": 0,
              },
              Object {
                "fixableCount": 0,
                "totalCount": 1,
              },
              Object {
                "fixableCount": 0,
                "totalCount": 5,
              },
              Object {
                "fixableCount": 0,
                "totalCount": 0,
              },
              Object {
                "fixableCount": 0,
                "totalCount": 1,
              },
              Object {
                "fixableCount": 0,
                "totalCount": 5,
              },
              Object {
                "fixableCount": 0,
                "totalCount": 52,
              },
            ],
          },
        },
      ],
      "_type": "sysdig_image_scan",
      "category": "Image Scan",
      "createdOn": undefined,
      "displayName": "debian sha256:43d28810c1b4c28a1be3bac8e0e40fcc472b2bfcfcda952544ed99cb874d2b1a",
      "fixablePackages": Array [],
      "id": "17144f2f45c295bc2fdcae5db51044ca",
      "internal": false,
      "metadata.architecture": "amd64",
      "metadata.author": "",
      "metadata.baseOS": "debian 11.5",
      "metadata.createdAt": 1663030579369,
      "metadata.digest": "debian@sha256:73060f9fdf46e142b63fddde45759f12aea093156445c32ab667338d0563f5c4",
      "metadata.imageId": "sha256:43d28810c1b4c28a1be3bac8e0e40fcc472b2bfcfcda952544ed99cb874d2b1a",
      "metadata.labels": null,
      "metadata.layersCount": 1,
      "metadata.os": "linux",
      "metadata.pullString": "debian",
      "metadata.size": 129313792,
      "name": "debian sha256:43d28810c1b4c28a1be3bac8e0e40fcc472b2bfcfcda952544ed99cb874d2b1a",
      "packageCount": 96,
      "packageTypes": Array [
        "os",
      ],
      "runningFixablePackages": null,
      "storedAt": 1663041226858,
      "summary": "Image Scan",
      "type": "dockerImage",
      "vulnsBySev": Array [
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
      ],
    },
    Object {
      "_class": Array [
        "Assessment",
      ],
      "_key": "sysdig_image_scan:17140adb60b1639a2f8d615f88de1bfa",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "fixablePackages": Array [
              Object {
                "exploitCount": 0,
                "id": "17140adb60b1639ad35b4c1a76a5fba9",
                "name": "libpcre2-8-0",
                "suggestedFix": "10.36-2+deb11u1",
                "type": "os",
                "version": "10.36-2",
                "vulnsBySev": Array [
                  0,
                  0,
                  2,
                  0,
                  0,
                  0,
                  0,
                  0,
                ],
              },
              Object {
                "exploitCount": 0,
                "id": "17140adb60b1639a3c88473e6e13338a",
                "name": "zlib1g",
                "suggestedFix": "1:1.2.11.dfsg-2+deb11u2",
                "type": "os",
                "version": "1:1.2.11.dfsg-2+deb11u1",
                "vulnsBySev": Array [
                  0,
                  0,
                  1,
                  0,
                  0,
                  0,
                  0,
                  0,
                ],
              },
              Object {
                "exploitCount": 0,
                "id": "17140adb60b1639a39c15c260c5fdaf2",
                "name": "libc6",
                "suggestedFix": "2.31-13+deb11u4",
                "type": "os",
                "version": "2.31-13+deb11u3",
                "vulnsBySev": Array [
                  0,
                  0,
                  0,
                  1,
                  0,
                  0,
                  0,
                  6,
                ],
              },
              Object {
                "exploitCount": 0,
                "id": "17140adb60b1639aaa5ddbdf428888b5",
                "name": "libc-bin",
                "suggestedFix": "2.31-13+deb11u4",
                "type": "os",
                "version": "2.31-13+deb11u3",
                "vulnsBySev": Array [
                  0,
                  0,
                  0,
                  1,
                  0,
                  0,
                  0,
                  6,
                ],
              },
            ],
            "id": "17140adb60b1639a2f8d615f88de1bfa",
            "isEVEEnabled": false,
            "labels": null,
            "metadata": Object {
              "architecture": "amd64",
              "author": "",
              "baseOS": "debian 11.4",
              "createdAt": "2022-08-23T00:20:40.700966019Z",
              "digest": "debian@sha256:d52921d97310d0bd48dab928548ef539d5c88c743165754c57cfad003031386c",
              "imageId": "sha256:dd8bae8d259fed93eb54b3bca0adeb647fc07f6ef16745c8ed4144ada4d51a95",
              "labels": null,
              "layersCount": 1,
              "os": "linux",
              "pullString": "debian",
              "size": 129216000,
            },
            "packageCount": 96,
            "packageTypes": Array [
              "os",
            ],
            "policyEvaluations": Array [
              Object {
                "creationTimestamp": "2022-09-02T11:58:59.03062Z",
                "evaluationResult": "failed",
                "failuresCount": Object {
                  "imageConfigCreationDate": 0,
                  "imageConfigDefaultUser": 0,
                  "imageConfigEnvVariable": 0,
                  "imageConfigInstructionIsPkgManager": 0,
                  "imageConfigInstructionNotRecommended": 0,
                  "imageConfigLabel": 0,
                  "imageConfigSensitiveInformationAndSecrets": 0,
                  "vulnDenyList": 0,
                  "vulnSeverityAndThreats": 13,
                },
                "id": "17140adb60b1639a9bc2d5b0f765e4d2",
                "identifier": "sysdig_best_practices",
                "name": "Sysdig Best Practices",
                "policyType": "alwaysApply",
                "updateTimestamp": "2022-09-02T11:58:59.03062Z",
              },
            ],
            "runningFixablePackages": null,
            "storedAt": "2022-09-12T07:01:39.742909338Z",
            "type": "dockerImage",
            "vulnsBySev": Array [
              Object {
                "fixableCount": 0,
                "totalCount": 0,
              },
              Object {
                "fixableCount": 0,
                "totalCount": 0,
              },
              Object {
                "fixableCount": 3,
                "totalCount": 4,
              },
              Object {
                "fixableCount": 2,
                "totalCount": 7,
              },
              Object {
                "fixableCount": 0,
                "totalCount": 0,
              },
              Object {
                "fixableCount": 0,
                "totalCount": 1,
              },
              Object {
                "fixableCount": 0,
                "totalCount": 5,
              },
              Object {
                "fixableCount": 0,
                "totalCount": 52,
              },
            ],
          },
        },
      ],
      "_type": "sysdig_image_scan",
      "category": "Image Scan",
      "createdOn": undefined,
      "displayName": "debian sha256:dd8bae8d259fed93eb54b3bca0adeb647fc07f6ef16745c8ed4144ada4d51a95",
      "fixablePackages": Array [
        Object {
          "exploitCount": 0,
          "id": "17140adb60b1639ad35b4c1a76a5fba9",
          "name": "libpcre2-8-0",
          "suggestedFix": "10.36-2+deb11u1",
          "type": "os",
          "version": "10.36-2",
          "vulnsBySev": Array [
            0,
            0,
            2,
            0,
            0,
            0,
            0,
            0,
          ],
        },
        Object {
          "exploitCount": 0,
          "id": "17140adb60b1639a3c88473e6e13338a",
          "name": "zlib1g",
          "suggestedFix": "1:1.2.11.dfsg-2+deb11u2",
          "type": "os",
          "version": "1:1.2.11.dfsg-2+deb11u1",
          "vulnsBySev": Array [
            0,
            0,
            1,
            0,
            0,
            0,
            0,
            0,
          ],
        },
        Object {
          "exploitCount": 0,
          "id": "17140adb60b1639a39c15c260c5fdaf2",
          "name": "libc6",
          "suggestedFix": "2.31-13+deb11u4",
          "type": "os",
          "version": "2.31-13+deb11u3",
          "vulnsBySev": Array [
            0,
            0,
            0,
            1,
            0,
            0,
            0,
            6,
          ],
        },
        Object {
          "exploitCount": 0,
          "id": "17140adb60b1639aaa5ddbdf428888b5",
          "name": "libc-bin",
          "suggestedFix": "2.31-13+deb11u4",
          "type": "os",
          "version": "2.31-13+deb11u3",
          "vulnsBySev": Array [
            0,
            0,
            0,
            1,
            0,
            0,
            0,
            6,
          ],
        },
      ],
      "id": "17140adb60b1639a2f8d615f88de1bfa",
      "internal": false,
      "metadata.architecture": "amd64",
      "metadata.author": "",
      "metadata.baseOS": "debian 11.4",
      "metadata.createdAt": 1661214040700,
      "metadata.digest": "debian@sha256:d52921d97310d0bd48dab928548ef539d5c88c743165754c57cfad003031386c",
      "metadata.imageId": "sha256:dd8bae8d259fed93eb54b3bca0adeb647fc07f6ef16745c8ed4144ada4d51a95",
      "metadata.labels": null,
      "metadata.layersCount": 1,
      "metadata.os": "linux",
      "metadata.pullString": "debian",
      "metadata.size": 129216000,
      "name": "debian sha256:dd8bae8d259fed93eb54b3bca0adeb647fc07f6ef16745c8ed4144ada4d51a95",
      "packageCount": 96,
      "packageTypes": Array [
        "os",
      ],
      "runningFixablePackages": null,
      "storedAt": 1662966099742,
      "summary": "Image Scan",
      "type": "dockerImage",
      "vulnsBySev": Array [
        0,
        0,
        3,
        2,
        0,
        0,
        0,
        0,
      ],
    },
  ],
  "encounteredTypes": Array [
    "sysdig_image_scan",
  ],
  "numCollectedEntities": 2,
}
`;
